qucs/qucs/qucs.cpp

```diff
@@ -834,44 +860,37 @@ void QucsApp::slotSelectComponent(QListWidgetItem *item)
   char *CompFile_cptr;
 
   if (!CompSearch->text().isEmpty()) {
-    //comp search is not empty, search all category to get a component in same name
-    QStringList cats = Category::getCategories();
-    int i = 0;
-    //search normal component
-    foreach (QString it, cats) {
-      Comps = Category::getModules (it);
-      foreach(Module *mod, Comps) {
-        if (mod->info) {
-          (*mod->info)(CompName, CompFile_cptr, false);
-          if (CompName == name) {
```

Comment:
yes, the problem was trying to identify a component via its description. This does not work when two components have the same description, like "n-MOSFET" for the 3-terminal MOSFET and the 3-terminal plus substrate connection MOSFET.
The context it the component search, which is activated by typing a string in the line input at the bottom of the Components tab on the left side of the GUI (see #543 description also)

Until now the icons in the Components tab did not directly identify a component; when a component icon was clicked the corresponding component was identified indirectly, looking at the currently selected component category in the Combo Box at the top and the icon index in the IconView pane.

What I do here is to put these information in the Icon data, so there is no need to search which component corresponds to the selected icon, now the icon itself holds these data and the (wrong) search for which component has the same description as the icon is no longer needed.